src/components/patch-assignment-form/components/rows.tsx

```diff
@@ -51,20 +51,54 @@ export const TableRow = ({
   );
   const [validationError, setValidationError] = useState<string | null>(null);
 
-  function onConfirmButtonClick() {
+  const handleConfirmClick = useCallback(() => {
     if (newOfficerEmail && !newOfficerEmail.includes("@hackney.gov.uk")) {
       setValidationError("Email address must include @hackney.gov.uk");
       return;
     }
     setValidationError(null);
     editPatchAssignment(areaOrPatch, newOfficerName, newOfficerEmail, handleSubmission);
-  }
-  function onCancelButtonClick() {
+  }, [
+    areaOrPatch,
+    newOfficerName,
+    newOfficerEmail,
+    handleSubmission,
+    setValidationError,
+  ]);
+
+  const handleCancelClick = useCallback(() => {
     setNewOfficerName(officer?.name || "");
     setNewOfficerEmail(officer?.contactDetails?.emailAddress?.toLowerCase().trim() || "");
     setReassigningPatch(null);
     setValidationError(null);
-  }
+  }, [
+    officer,
+    setReassigningPatch,
+    setNewOfficerName,
+    setNewOfficerEmail,
+    setValidationError,
+  ]);
+
+  const handleOfficerNameChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setNewOfficerName(event.target.value.trim());
```

Comment:
this value will never be empty null?

Reply:
yeah it can be but it doesn't cause an error - i've tested this out